File: src/graphics/cpp/movable_image/main_window.cpp
#include "main_window.h"
#include "movable_image.h"
#include "ui_main_window.h"

MainWindow::MainWindow(QWidget *parent)
    : QMainWindow(parent), ui(new Ui::MainWindow) {
  ui->setupUi(this);
  scene = new QGraphicsScene(this);
  ui->graphicsView->setScene(scene);

  // set QSurfaceFormat for graphicsView

  auto text = scene->addText("Drop pictures here", QFont("Ubuntu", 20));
  text->setFlag(QGraphicsItem::ItemIsMovable);
  text->setFlag(QGraphicsItem::ItemIsSelectable);
}

MainWindow::~MainWindow() { delete ui; }

void MainWindow::on_addButton_released() {
  scene->addItem(new MovableImage(this));
}

void MainWindow::on_removeButton_released() {
  for (auto item : scene->selectedItems()) {
    scene->removeItem(item);
    delete item;
  }
}

File: src/graphics/python/movable_image/main_window.py
from movable_image import MovableImage
from PyQt6 import QtGui, QtWidgets, uic
from PyQt6.QtWidgets import QGraphicsItem


class MainWindow(QtWidgets.QMainWindow):
    def __init__(self):
        super().__init__()
        uic.loadUi("main_window.ui", self)
        self.scene = QtWidgets.QGraphicsScene(self)
        self.graphicsView.setScene(self.scene)
        text = self.scene.addText("Drop pictures here", QtGui.QFont("Ubuntu", 20))
        text.setFlag(QGraphicsItem.GraphicsItemFlag.ItemIsMovable)
        text.setFlag(QGraphicsItem.GraphicsItemFlag.ItemIsSelectable)
        self.show()

    def on_addButton_released(self):
        new_item = MovableImage(self)
        self.scene.addItem(new_item)

    def on_removeButton_released(self):
        for item in self.scene.selectedItems():
            self.scene.removeItem(item)
            del item
